src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -457,51 +456,42 @@ private static void ValidateAccessor(Expression instance, MethodInfo method, Par
 
         private static void ValidateAccessorArgumentTypes(MethodInfo method, ParameterInfo[] indexes, ref ReadOnlyCollection<Expression> arguments)
         {
-            if (indexes.Length > 0)
+            if (indexes.Length != arguments.Count)
```

Comment:
There are no tests covering some of the paths being changed here.  Before we go trying to change code like this, at a minimum anything being changed needs test coverage.